Remove columns and ones from slow2d transpose signatures

jbschlosser opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69044
* #68899
* __->__ #68898
* #68897

Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

jbschlosser committed:
Remove columns and ones from slow2d transpose signatures

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

zou3519 commented on aten/src/ATen/native/NaiveConvolutionTranspose2d.cpp:
```diff
@@ -281,20 +276,12 @@ void slow_conv_transpose2d_out_cpu_template(
-  // Note: this buffer can be shared with other modules, it only ever gets
-  // increased, and always contains ones.
-  if (ones.dim() != 2 ||
-      ones.size(0) * ones.size(1) < output_height * output_width) {
-    // Resize plane and fill with ones...
-    ones.resize_({output_height, output_width});
-    ones.fill_(1);
-  }
+  Tensor ones = bias.defined() ? at::ones({output_height, output_width}, input_.options()) : Tensor();
```
Hmm, is there a performance impact on this? I remember we benchmarked the slow conv finput changes, but did we benchmark this one as well?

zou3519 replied:
It looks like we were creating column and ones on the fly previously anyways, so this shouldn't have a performance impact. Is this a correct assessment?

zou3519 reviewed (approved):
LGTM if my understanding is right

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser committed:
Update on "Remove columns and ones from slow2d transpose signatures"


Differential Revision: [D32655873](https://our.internmc.facebook.com/intern/diff/D32655873)

[ghstack-poisoned]

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).

jbschlosser commented:
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32655873).
